openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -538,26 +538,46 @@ def run(self, *, dry=False, verbose=True,
             has_solvent=solvent_comp is not None,
         )
 
+        # workaround for conformer generation failures
+        # see openfe issue #576
+        # calculate partial charges manually if not already given
+        # convert to OpenFF here,
+        # and keep the molecule around to maintain the partial charges
+        off_small_mols = {
+            'stateA': mapping.componentA.to_openff(),
+            'stateB': mapping.componentB.to_openff(),
+            'both': [m.to_openff() for m in small_mols
+                     if (m != mapping.componentA and m != mapping.componentB)]
+        }
+
         # b. force the creation of parameters
         # This is necessary because we need to have the FF generated ahead of
         # solvating the system.
         # Note: by default this is cached to ctx.shared/db.json so shouldn't
         # incur too large a cost
         self.logger.info("Parameterizing molecules")
-        for comp in small_mols:
-            offmol = comp.to_openff()
-            system_generator.create_system(offmol.to_topology().to_openmm(),
-                                           molecules=[offmol])
-            if comp == mapping.componentA:
-                molB = mapping.componentB.to_openff()
-                system_generator.create_system(molB.to_topology().to_openmm(),
-                                               molecules=[molB])
+        for mol in chain([off_small_mols['stateA'], off_small_mols['stateB'],
+                         *off_small_mols['both']]):
+            # robustly calculate partial charges;
+            if mol.partial_charges is not None and np.any(mol.partial_charges):
+                # skip if we have existing partial charges
```

Comment:
```suggestion
                # skip if we have existing partial charges unless they are zero (see openmmforcefields)
```

sorry I'm on phone so terrible formatting, but something like this would be great so we remember why